### ISSUE TITLE
Residences @ Daniel Webster 123-E0679

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#931]
In GitLab by @kdjstudios on Jun 6, 2018, 13:33

**Submitted by:** "Kimberly Gagner" <kimberly.gagner@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2018-06-06-12351

**Server:** Internal

**Client/Site:** Billerica

**Account:** 123-E0679

**Issue:** 

I had noticed the account patching activity was not active so I have attempted to reactivate it and it will not activate.  So, I went in and added a new field for patching and when I went to check that one is also not active. 

Can you tell me why it will not allow me to add this billing code or activity in this account?  Please try and let me know what happens when you try.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2018, 13:34

Kim also wrote:

> I have been in a bunch of other accounts and I am noticing all the "patching" activity fields are deactivated ? 
> 
> It appears it might be all of the patching fees, please advise.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2018, 13:44

@smasih @pchaudhary @tim.traylor 

I wrote:

> First: In response to your ability to reactivate fees, we would like to confirm the steps you are performing and the error you are receiving? We will also include the steps below in case you are performing them incorrectly.
> 
> 1. Locate account and click "Edit Account" button.
> 2. Find the deactivated activity (will be grayed out).
> 3. Click the "i" button next to this field and click "OK" on the pop up to confirm to reactivate. 
> 4. Click the "Save" button at the bottom of the account edit page to save your changes. (Not the "Save & Next" button)
> 
> Please let us know if you are still unable to activate the fee after following these steps; as it may be related to what is addressed below in the second topic. 
> 
> Second: In response to multiple accounts having the "Patching" activity deactivated. We quickly looked at the site's fees and it would appear there are duplicates of each activity, with once being activated and one being deactivated. Which if the deactivated one is the one on the accounts this would cause this. We have informed the support team on this to see if this is having any effect on the above too.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2018, 14:53

Kim wrote:

> Yes, I did press on the "I" then a message pops up asking if I wish to reactivate which I choose "ok" then the spot where you can write the description turns yellow (you cannot type anything in there) and nothing happens it is still not active once you save and go back and check.  
> 
> I looked at a few other accounts that I noticed were off  123-E0137 & 123-E0667.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 7, 2018, 05:40

Hi [Kyle] (@kdjstudios ),  

Please remember to ask as more detail as possible from the client for deep investigation. 
As we don't have any idea about the activity he is trying to activate but after looking into the account we guess that he was trying for the `123 Patch Minutes 0501` activity, so below are the finding of ours:

*  The activity must be activated on the Site level first to activate the same on the account level. But the case seems here that the activity was deactivated on the Site level when he was trying to activate on the account level.


Please confirm with Kimberly to activate it on the Site level first.
We'll wait for the response and let us know if there's something else we need to do on this ticket.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 7, 2018, 05:40

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 7, 2018, 09:52

@pchaudhary @rchauhan @tim.traylor 

As stated in my response above. There are duplicate activities, one activated and one deactivated. both with the same codes...?? 
- So which code is actually being used? 
- How is operations able to add a deactivated code at the site level to an account?
- Should we have validation here?
- Should we have warnings or alerts telling the user the code needs to be activated at the site level; rather then saying saved successfully?
- I highlighted the active 0501 code below in the screen shot. 

![image](/uploads/ef78c0159a9148bfb3273d032a5bfe32/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 8, 2018, 24:59

> So which code is actually being used?

Only activated activity (on site level) will be used.



> How is operations able to add a deactivated code at the site level to an account?

Because, there is no code in place to stop user from doing that. If we stop giving an option for deactivated activities, it will lead to further issues.




> Should we have validation here?

We, would suggest no, its not required. If there is an active activity selected on account, and later it's deactivated on site, so if there is a validation in place system will remove the activity when the next time page loads. 
Secondly, we need to preserve rate changes.



> Should we have warnings or alerts telling the user the code needs to be activated at the site level; rather then saying saved successfully?

Agreed, this is what we should be doing, perfect. We will discuss it with Tim before starting development on it.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 8, 2018, 24:59

`estimate 1hr`

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 9, 2018, 03:58

@kdjstudios 

sorry, I forgot to tag you in my previous comment. Can you please check this http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/931#note_21610

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 9, 2018, 16:39

@smasih @pchaudhary @tim.traylor 

I would seem we need validation to prevent this from happening. Please discuss with @tim.traylor as you mentioned in your comment. This will also probably need approval from Gary as it deal with how operations will use the system.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 14, 2018, 06:50

related #892

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 18, 2018, 11:08

@smasih @pchaudhary @rchauhan 

Kimberly wrote:

> I am checking do you have any information on why this was happening?  I wanted to make sure this was resolved before my next billing cycle as I have several clients who are billed for patching.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 18, 2018, 11:08

Increasing priority, as this is effecting clients bills.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 19, 2018, 01:23

@kdjstudios 

Adding the validation is a way to inform user, but there is no bug in the system during activity processing.

We should help Ops to understand how activity works when activated or deactivated on both Site level and Account level.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 19, 2018, 08:24

@smasih 

When at the account level you can add a code that has been deactivated at the site level. This is an issue!Either of two things need to happen:

- Display "Active" or "Inactive" in front of the code on the drop down list.
- Do not display Inactive codes at all in the drop down list.


![image](/uploads/49e68ced1f9f62e85b19d33181163354/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 20, 2018, 01:33

@kdjstudios 

on saving the account activity it gets greyed out if it's deactivated on site level.



> - Display "Active" or "Inactive" in front of the code on the drop down list.
>
> - Do not display Inactive codes at all in the drop down list.

I understand what you are saying, but these names are being displayed by using a default functionality, altering that will either require more time or we need to code again the whole section.

As per our understanding, "showing some notification on selecting deactivated activity on site level" will take relatively less time.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 20, 2018, 09:57

@smasih 

Please provide time estimates for both solutions. We will discuss this with Gary and @tim.traylor tomorrow to approve one of the soltuions.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 21, 2018, 02:27

@kdjstudios @tim.traylor 

As per our discussion with Tim yesterday (20^th Jun) we agreed to work on the code to implement following options and we can ask Gary to choose any of the option:

* differentiate inactive activity by using different color  
![Screenshot-9](/uploads/c590c716547c96214b4afa625e186258/Screenshot-9.png)

* append a text saying that it's inactive.   
![Screenshot-11](/uploads/07e66252cf7a129930e38930889e4fe0/Screenshot-11.png)

* or both


**completed work on this ticket, just waiting for an approval**

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 21, 2018, 06:55

On investigating further, we found a way to hide the deactivated activities on account form.  
So, this is also included as an option adding on to the Sumeet's previous comment.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 21, 2018, 10:24

Per discussion today, Updating Kim that this will be resolved on this release.

> The support team was able to research this and has found the this is due to multiple activities (Active and Inactive) at the site level using the same code number. The support team has currently found a solution to no display any of the inactive activities at the account level in the drop down in order to prevent adding inactive activities to accounts. This functionality is currently being tested and should be released to production on June 23rd. We will send another update once this has been released.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 23, 2018, 05:36

@kdjstudios 

During our call with Tim, we were informed that Kim will be adding a details on the Saturday morning. So, we released this change.  

Only active fees/activities will come as an option during the selection of activity/recurring fees on the account level.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 25, 2018, 08:53

Hi [Kyle] (@kdjstudios ),   

We found an error after the release of this change. So, we had to revert back.

This fix is partially released.

Working on account edit page. (Showing only active fees/activities)

Not working on New account page (Showing only **all** fees/activities)

> - Not working on New account page (Showing only **all** fees/activities)

this will be resolved by tomorrow

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 25, 2018, 09:02

@pchaudhary 

Thank you, please keep us updated.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 26, 2018, 08:15

@kdjstudios 

fixed on production. (both)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2018, 08:27

@pchaudhary 

Thanks for the update. Looks good. marking for close

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 29, 2018, 11:41

closed
